### PR TITLE
fix(batch,agg): support all types for batch single_value, fixing a bug found by sqlsmith

### PR DESCRIPTION
**Pull request by richardchien**
Signed-off-by: Richard Chien <stdrc@outlook.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Fix https://buildkite.com/singularity-data/pull-request/builds/6821#0182d365-16bc-49c4-a71a-326feeef66c4/5162-5241

#4864

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/expr/src/vector_op/agg/aggregator.rs [hunk 1]
```diff
@@ -215,16 +215,17 @@ pub fn create_agg_state_unary(
...
         (Count, count, date, int64, Some(0)),
         (Count, count, timestamp, int64, Some(0)),
         (Count, count, time, int64, Some(0)),
+        (Count, count_str, varchar, int64, Some(0)),
         (Count, count_struct, struct_type, int64, Some(0)),
         (Count, count_list, list, int64, Some(0)),
         (Sum, sum, int16, int64, None),
         (Sum, sum, int32, int64, None),
+        (Sum, sum, int64, int64, None),
         (Sum, sum, int64, decimal, None),
```
Comment (neverchanje) [inline src/expr/src/vector_op/agg/aggregator.rs:229]:
```suggestion
        (Sum, sum, int64, int64, None),
        (Sum, sum, int64, decimal, None),
```

Are the two signatures duplicated?

Comment (neverchanje) [reply src/expr/src/vector_op/agg/aggregator.rs:229]: Ok, looks like it was moved from a previous line. Just curious why there are two SUM for i64.

Comment (richardchien) [reply src/expr/src/vector_op/agg/aggregator.rs:229]: Seems mysterious...I think it's never called.

Comment (neverchanje) [reply src/expr/src/vector_op/agg/aggregator.rs:229]: If deleting it will not cause any testing failure, then just remove it :)

**Comment by neverchanje** [comment]
Since this pr fixed a bug caused by sqlsmith, please add a e2e test that can reproduce the bug.

File: src/expr/src/vector_op/agg/aggregator.rs (hunk 1 above)
Comment (richardchien) [reply src/expr/src/vector_op/agg/aggregator.rs:229]: Oh, we can't remove it. Found that `int64 + int64 => decimal` https://github.com/singularity-data/risingwave/blob/62cc3ebb17c38b3800ef666f676d00b9fef22b60/src/frontend/src/expr/agg_call.rs#L125

Comment (neverchanje) [reply src/expr/src/vector_op/agg/aggregator.rs:229]: Yes. That's what I mean.

Comment (richardchien) [reply src/expr/src/vector_op/agg/aggregator.rs:229]: The test broke after remove that line. It's used as global phase of `count`.

**Review by BugenZhao — APPROVED** [review]

**Comment by BugenZhao** [comment]
Seems like we've disabled `single_value(interval)` in sqlsmith (#4865) 🤔

**Comment by BugenZhao** [comment]
> Since this pr fixed a bug caused by sqlsmith, please add a e2e test that can reproduce the bug.

I believe the issue is clear enough? As we forgot to dispatch the `single_value` for several data types.

**Comment by richardchien** [comment]
Think we can leave this unmerged for now. Seems the existence of `single_value` still need some discussion according to #4866.

File: src/expr/src/vector_op/agg/aggregator.rs [hunk 2]
```diff
@@ -268,7 +268,18 @@ pub fn create_agg_state_unary(
         (SingleValue, SingleValue::new(), float64, float64, None),
         (SingleValue, SingleValue::new(), decimal, decimal, None),
         (SingleValue, SingleValue::new(), boolean, boolean, None),
+        (SingleValue, SingleValue::new(), date, date, None),
+        (SingleValue, SingleValue::new(), timestamp, timestamp, None),
+        (SingleValue, SingleValue::new(), time, time, None),
```
Comment (xiangjinwu) [inline src/expr/src/vector_op/agg/aggregator.rs:273]: FYI there are 2 more types: interval and timestampz
https://www.risingwave.dev/docs/latest/sql-data-types/

**Comment by lmatz** [comment]
As https://github.com/risingwavelabs/risingwave/pull/5562 removes `single_value` now, shall we close this

**Comment by richardchien** [comment]
> As #5562 removes `single_value` now, shall we close this

Sure

**Pull request closed without merging**